src/components/modals/revision-diff.jsx

```diff
@@ -0,0 +1,41 @@
+import PropTypes from "prop-types";
+import React from "react";
+import ReactDiffViewer from "react-diff-viewer";
+import styled from "react-emotion";
+
+const DiffContainer = styled("div")`
+  overflow: auto;
+`;
+
+export class RevisionDiff extends React.Component {
+  static propTypes = {
+    gettingRevisionContent: PropTypes.bool,
+    currentRevisionContent: PropTypes.string,
+    previousRevisionContent: PropTypes.string
+  };
+
+  render() {
+    if (
+      this.props.gettingRevisionContent ||
```

Comment:
this component should be connected to the store directly. as with the history list, the store knows everything that this component needs to render, and by connecting it directly we'll end up with more simple mapStateToProps functions, which is better than fewer complex ones.

also, i think that `gettingRevisionContent` is not actually needed. If the only states we choose to display i this component are (a) gettin revisions (b) no diff (c) some diff, then we can deduce that we are in the getting revisions state if either  `prevRev===undef || currRev===undef`. so I think we can remove all references to this variable.

(note tho that if you wish to admit case  (d) error getting revisions, then it might make sense to look at a `revisionContentFetchStatus` variable, as i suggest elsewhere)

Reply:
Again, I don't think connecting this to the redux store would improve things. You may be right that we could infer the fetching status in some other way though, i'll look into it.

Reply:
again, connecting components to the store is the convention we've adopted, and our experience has shown that it has many advantages. please give it a shot :-)